components/Chart.tsx

```diff
@@ -187,12 +187,12 @@ const ExpandableBarChartByPattern = (props: ExpandableBarChartByPatternProps) =>
                 {term.type === "Obligation" && term.strength > 0 && <PatternIcon size="24" className="ml-2 text-black" pattern={{ iconUrl: term.patternIconUrl }}/>}
               </div>
               {showLabels &&
-                <div className="flex-1 h-10 text-black text-sm flex items-center mr-3 text-right" style={{ gridColumn: `span ${8 - term.strength}` }}>
+                <div className="flex-1 h-10 text-black text-xs sm:text-sm flex items-center mr-3 text-right" style={{ gridColumn: showLabels ? `span ${8 - term.strength}` : `span ${5 - term.strength}` }}>
                   {term.type === "Obligation" && term.name}
                 </div>
               }
             </div>
-            <div className="flex-1 grid" style={{ gridTemplateColumns: `repeat(8, minmax(0, 1fr))`, direction: "ltr" }}>
+            <div className="flex-1 grid" style={{ gridTemplateColumns: showLabels ? `repeat(8, minmax(0, 1fr))` : `repeat(5, minmax(0, 1fr))`, direction: "ltr" }}>
```

Comment:
nit: It might be a bit neater to set a variable `colNums` (or a better name!)

```ts
const colNums = showLabels ? 8 : 5
return (
  <div 
    className="flex-1 grid" 
    style={{ 
      gridTemplateColumns: `repeat(${colNums}, minmax(0, 1fr))`, 
      direction: "ltr" 
    }}
  >
)
```